library/psa_crypto.c

```diff
@@ -3395,6 +3395,490 @@ psa_status_t psa_aead_decrypt( mbedtls_svc_key_id_t key,
     return( status );
 }
 
+/* Helper function to get the base algorithm from its variants. */
+static psa_algorithm_t psa_aead_get_base_algorithm( psa_algorithm_t alg )
+{
+    return PSA_ALG_AEAD_WITH_DEFAULT_LENGTH_TAG( alg );
+}
+
+/* Set the key for a multipart authenticated encryption operation. */
+psa_status_t psa_aead_encrypt_setup( psa_aead_operation_t *operation,
+                                     mbedtls_svc_key_id_t key,
+                                     psa_algorithm_t alg )
+{
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+    psa_status_t unlock_status = PSA_ERROR_CORRUPTION_DETECTED;
+    psa_key_slot_t *slot = NULL;
+
+    if( !PSA_ALG_IS_AEAD( alg ) || PSA_ALG_IS_WILDCARD( alg ) )
```

Comment:
I think this comment is still valid. `psa_aead_encrypt_setup()` and `psa_aead_decrypt_setup()` code is almost the same thus it would be beneficial to have a `psa_aead_setup()` function with a signature similar to that of `psa_cipher_setup()`: 
```C
static psa_status_t psa_aead_setup( psa_aead_operation_t *operation,        
                                      mbedtls_svc_key_id_t key,                 
                                      psa_algorithm_t alg,                      
                                      mbedtls_operation_t aead_operation )
```

and `psa_aead_encrypt/decrypt_setup()` just calling `psa_aead_setup()`.